porebuilder/tests/test_porebuilder.py

```diff
@@ -50,3 +53,35 @@ def test_particles_in_box(self, GraphenePoreSolvent):
             assert particle.xyz[0][0] < box.maxs[0]
             assert particle.xyz[0][1] < box.maxs[1]
             assert particle.xyz[0][2] < box.maxs[2]
+
+    def test_functionalize_surface(self):
+
+        class H(mb.Compound):
+            def __init__(self):
+                super(H, self).__init__()
+
+                self.add(mb.Particle(name='H'))
+                up_port = mb.Port(
+                    anchor=self[0], orientation=[
+                        0, 1, 0], separation=.075)
+                self.add(up_port, "up")
+
+        class O(mb.Compound):
+            def __init__(self):
+                super(O, self).__init__()
+
+                self.add(mb.Particle(name='O'))
+                up_port = mb.Port(
+                    anchor=self[0], orientation=[
+                        0, 1, 0], separation=.075)
+                self.add(up_port, "up")
+
+        with pytest.raises(ValueError):
+            pb.GraphenePoreFunctionalized(
+                func_groups=[H(), O()], func_percent=[.03, .04, .03])
+
+        for per in range(1, 10):
+            pore = pb.GraphenePoreFunctionalized(
```

Comment:
* This is a fairly long test, maybe switch it to only 3 values of `per`?

* We should test this for different values of `pore_width` to verify the process of finding the surfaces. Two or three values should be enough